### PR TITLE
feat(KEN-444): Update now carries the kendex command across

### PR DESCRIPTION
**Pull request by bmethod**
The app's Update now replaced the AppImage and relaunched, and never touched the
kendex command beside it. On an install.sh machine the two halves are two files,
so a person who updated from the app was left running terminal automation on the
old command while the app moved on.

`kendex update` already moved both halves on a direct install. The app did not.

## One path, two callers

The command-replacement logic moved out of `crates/cli` into
`crates/core/src/command_update.rs`, and both ends call it. A second
implementation of "replace the kendex command" is how the two ends drift, which
is the defect this issue exists to prevent — so there is one.

It carries KEN-754's guarantee with it: the command binary is verified under the
pinned key before it replaces anything, on the app-driven path as much as the
CLI one.

## What the issue asked for

- The plugin swaps the app and the same flow carries the command across; a
  partial failure is reported rather than silent.
- Package-managed installs still self-update neither half — `allow_replacement`
  already refused anything but a direct install and still does. An install with
  no command beside the app stays app-only; absence is not failure.
- Tests pin the two versions equal after a family update, and cover skew in both
  directions.

## Note on the title

The issue is titled "install.sh family updates". It is not built there:
`docs/plans/update-notice-and-account.md` records that app-and-CLI-together
lives in `kendex update`, and this change makes the app reuse that path rather
than adding a third place where an update can happen.

## Rebase

Rebased onto main after #1813 landed. That PR added a `curl_args` test to the
CLI's update tests; this change moved `curl_args` into core, so the test moved
with it and still asserts what it did — worth more there, since the function now
serves both update paths.

`tools/guard` green by exit code.

Closes KEN-444

### CODE REVIEW

File: crates/core/src/command_update.rs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    feed_url: &str,
+    release: &str,
+    target: &str,
+    public_key: &str,
+) -> Result<CommandHalf, String> {
+    let CommandBeside::Ours(path) = beside else {
+        return Ok(CommandHalf::Untouched);
+    };
+    let feed = ReleaseFeed::parse(&fetch(feed_url)?).map_err(|error| error.to_string())?;
+    if !one_release(&feed.version, release) {
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:230]: Tracked: KEN-820

Right, and it is the same gap two reviewers found on #1815: the signature proves the bytes and not which release they are, so a feed advertising the signed manifest's release while pointing `assets[target]` at an older legitimately signed command passes verification and downgrades. Comparing a version the attacker supplies is what makes it work.

Binding version, target and asset needs signed feed metadata, which is a mechanism this PR should not grow — it exists to stop the app and the command drifting apart, and it reuses the CLI's verified path rather than adding a second one. KEN-820 carries the binding with the downgrade and wrong-target cases as its tests.

File: crates/app/src/app_update.rs [hunk 2]
```diff
@@ -125,10 +129,67 @@ fn manifest_endpoint() -> Result<tauri::Url, String> {
...
+fn command_beside(
+    env: &Env,
+    install: &AppInstall,
+    path_var: Option<&std::ffi::OsStr>,
+) -> CommandBeside {
+    command_beside_app(
+        &Host,
+        &command_candidates(&env.home, path_var),
+        install.judged_path(),
+    )
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/app/src/app_update.rs:240]: Fixed in 093d3f284

The running executable is excluded independently of the updater's judged path, so a Windows install whose directory is on `PATH` cannot classify the app as the command it is about to replace.

File: crates/core/src/command_update.rs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    probe: &dyn HostProbe,
+    candidates: &[PathBuf],
+    app_bytes: Option<&Path>,
+) -> CommandBeside {
+    let app_bytes = app_bytes.map(|path| probe.resolve(path));
+    for candidate in candidates {
+        if !probe.exists(candidate) {
+            continue;
+        }
+        let resolved = probe.resolve(candidate);
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Fixed in 093d3f284

Every `PATH` and install-directory candidate goes through an executable-file probe now, so a directory or a non-executable file named `kendex` is skipped the way a shell would skip it, rather than being written over.

File: crates/core/src/command_update.rs [hunk 4]
```diff
@@ -0,0 +1,272 @@
+//! Bringing an installed kendex binary to a release: the download, the
+//! signature check, the write, and the command half of a family update.
+//!
+//! Both shells drive one release across the desktop app and the `kendex`
+//! command, and both write the half that is their own state marker last.
+//! For `kendex update` that is the command, whose baked version the next
+//! run reads to decide whether it is done; for the app it is the app,
+//! whose baked version the sidebar card reads. Either way a failure before
+//! the marker leaves both halves where they were, so the next attempt
+//! repeats them rather than stopping at already-current.
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Fixed in 093d3f284

The claim was false and the changelog one was consumer-facing, which is the worse half. Both now describe what happens: a split is possible, it is reported rather than hidden, and the marker is written last so a retry moves both halves. That is the property the code actually has, and `app_half_failed` exists because of it.

File: changelog.d/changed/KEN-444.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+- The app's **Update now** brings the `kendex` command across with it, so no
+  terminal is left on the old release. Neither half moves unless both can, and
```
[the remark this answers is not in the document]
Comment (bmethod) [reply changelog.d/changed/KEN-444.md]: Fixed in 093d3f284

The claim was false and the changelog one was consumer-facing, which is the worse half. Both now describe what happens: a split is possible, it is reported rather than hidden, and the marker is written last so a retry moves both halves. That is the property the code actually has, and `app_half_failed` exists because of it.

File: changelog.d/changed/KEN-444.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+- The app's **Update now** brings the `kendex` command across with it, so no
+  terminal stays on the old release. If one half lands without the other the
```
[the remark this answers is not in the document]
Comment (bmethod) [reply changelog.d/changed/KEN-444.md]: Fixed in 49c653239

You were right that the claim was unconditional and the behaviour is not. It was fixed at the source rather than in the prose: a `NotOurs` command now carries the channel that owns it, the card names that command and the command that will update it, and the changelog and the ARCHITECTURE sentence describe that instead of promising both halves always move.

The silent case was the real defect — an app that updates, restarts and clears its card while the terminal command stays old is acceptable only if it says so, and it did not.

File: crates/core/src/command_update.rs [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+/// `~/.local/bin`.
+pub fn command_candidates(home: &Path, path_var: Option<&OsStr>) -> Vec<PathBuf> {
+    let on_path = path_var
+        .into_iter()
+        .flat_map(std::env::split_paths)
+        .map(|dir| dir.join(COMMAND_NAME));
+    let install_sh = [
+        home.join(".local").join("bin").join(COMMAND_NAME),
+        Path::new("/usr/local/bin").join(COMMAND_NAME),
+    ];
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:110]: Fixed in 49c653239

Confirmed: `install.sh` and `command_candidates` held the same two roots, and the test asserted the same literals, so nothing could have noticed a divergence. The roots have one spelling now, as named constants, and contract tests in the existing install-script suite hold the installer's own destination selection against them rather than against a copy.

File: crates/core/src/command_update.rs [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+    let Some(asset) = feed.asset_for(target) else {
+        return Err(format!(
+            "release {release} publishes no kendex command for {target}, so the command installed here would be left behind; nothing was updated"
+        ));
+    };
+    // Named as the command half. Read off a card that has just been
+    // pressed, a bare fetch or signature error says nothing about which of
+    // the two halves it came from.
+    download(asset)
+        .and_then(|command| command.install_at(path, public_key))
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Tracked: KEN-820

Same gap, now raised by a fourth reviewer across three PRs: the signature proves the bytes and not which release they are, so a feed labelling the current release while pointing at an older legitimately signed command verifies, and the split you describe follows.

Binding the artifact to its release and target needs signed feed metadata. That is a mechanism this PR should not grow — it exists to stop the app and the command drifting apart and it reuses the CLI's verified path rather than adding a second one, and it is under a scope freeze after its diff doubled. KEN-820 carries the binding, with the downgrade and wrong-target cases as its tests, and it is queued next in this lane.

File: crates/core/src/command_update.rs [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+        if !probe.is_command(candidate) {
+            continue;
+        }
+        let resolved = probe.resolve(candidate);
+        if running.contains(&resolved) {
+            continue;
+        }
+        return match for_cli(&resolved, probe) {
+            InstallChannel::Direct => CommandBeside::Ours(resolved),
+            owned => CommandBeside::NotOurs(owned),
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Fixed in cdbb4ae87

Premise confirmed: `for_cli` established replaceability and package-ownership, never ownership. An executable someone wrote and named `kendex` in a writable directory classified as `Direct` and would have been overwritten.

No record existed to read, so the installers now write one: `install.sh` records the path it installed, and `kendex update` does the same when it replaces the command. Only a path an installer recorded is ours to replace; anything else takes the `NotOurs` path, where the card names the command and says it will not be carried. The candidate is never executed to identify it.

Refusing to touch a file we cannot prove is ours is the right default here — it costs a person one command, and the alternative costs them their own script.

File: crates/core/src/command_update.rs [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+        let resolved = probe.resolve(candidate);
+        if running.contains(&resolved) {
+            continue;
+        }
+        return match for_cli(&resolved, probe) {
+            // Asked before the record, because a package manager's copy
+            // has an owner worth naming and the card can name it. Only
+            // the arm that would replace the file needs proof.
+            InstallChannel::Direct if installed.as_deref() == Some(resolved.as_path()) => {
+                CommandBeside::Ours(resolved)
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Fixed in c8d81d730

Both were the record being written and then not trusted, so they were fixed together. `command_beside_app` now asks package ownership first, then the record proves the file, then writability decides.

A recorded path this app cannot write is its own state, `NeedsPrivilege`, rather than `NotOurs` — the card names the command and the `sudo kendex update` that supplies what is missing. The recorded path is chained onto the candidate list last, so `~/.cargo/bin/kendex` under a Finder launch is found while `PATH` still decides which copy wins where several exist.

`for_cli` was split so `package_owner` answers ownership alone: one `Unknown` had been meaning both "a manager owns it" and "nobody owns it and we cannot write it", which is what let a `sudo`-written `/usr/local/bin` install read as unowned.

File: crates/core/src/command_update.rs [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+/// `~/.local/bin`.
+pub fn command_candidates(home: &Path, path_var: Option<&OsStr>) -> Vec<PathBuf> {
+    let on_path = path_var
+        .into_iter()
+        .flat_map(std::env::split_paths)
+        .map(|dir| dir.join(COMMAND_NAME));
+    let install_sh = [
+        home.join(INSTALLER_HOME_BIN).join(COMMAND_NAME),
+        Path::new(INSTALLER_SYSTEM_BIN).join(COMMAND_NAME),
+    ];
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:110]: Fixed in c8d81d730

Both were the record being written and then not trusted, so they were fixed together. `command_beside_app` now asks package ownership first, then the record proves the file, then writability decides.

A recorded path this app cannot write is its own state, `NeedsPrivilege`, rather than `NotOurs` — the card names the command and the `sudo kendex update` that supplies what is missing. The recorded path is chained onto the candidate list last, so `~/.cargo/bin/kendex` under a Finder launch is found while `PATH` still decides which copy wins where several exist.

`for_cli` was split so `package_owner` answers ownership alone: one `Unknown` had been meaning both "a manager owns it" and "nobody owns it and we cannot write it", which is what let a `sudo`-written `/usr/local/bin` install read as unowned.

File: crates/core/src/command_update.rs (hunk 10 above)
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs]: Fixed in c8d81d730

You were right that a pathname is not durable proof. The record is now a path and a SHA-256, both required: `install.sh` writes both, `kendex update` records the running binary's bytes and re-records after a replacement lands, and the app's command half re-records what it installed. A different file at the same path fails the digest and reads as unproven. The candidate is never executed — the digest is read and hashed.

One limitation recorded rather than built around: after `sudo kendex update` the record refreshes under root's data directory, not the invoking user's, so that user's record goes stale and the app then reads the command as unproven. That is fail-closed — it refuses rather than overwrites — and any unprivileged `kendex update` restores it.

File: docs/ARCHITECTURE.md [hunk 12]
```diff
@@ -292,12 +292,12 @@ lives in one capability table read by core and UI.
...
   zero. Every read the app starts with runs beside the others.
 - **Discovery is the unsigned feed; both halves verify under one pinned key.**
-  Off the launch path, one cross-process transaction reads the feed at most
-  every six hours, keeps the last result beside any error, and never follows
-  the final link; the preference gates automatic contact; in both shells debug
-  builds alone honor `KENDEX_UPDATE_FEED`. Replacing needs the path behind the
-  running one writable, outside a system prefix; a package prefix names its
-  command instead, anything else neither. One sidebar card offers whichever.
+  Off the launch path, one cross-process transaction reads the feed every six
+  hours, keeps the last result and error, following no final link; the
```
[the remark this answers is not in the document]
Comment (bmethod) [reply docs/ARCHITECTURE.md]: Fixed in c8d81d730

Restored to "at most every six hours". Verified: the app schedules one startup check and the TTL suppresses another fetch, so an app left open does not poll.

File: crates/core/src/command_update.rs [hunk 13]
```diff
@@ -0,0 +1,378 @@
...
+        if let Some(owner) = package_owner(&resolved, probe) {
+            return CommandBeside::NotOurs(owner);
+        }
+        let ours = record.as_ref().is_some_and(|(path, digest)| {
+            *path == resolved && probe.digest(&resolved).as_deref() == Some(*digest)
+        });
+        if !ours {
+            // Nothing proves it ours. Refusing costs a person one command;
+            // being wrong costs them their file.
+            return CommandBeside::NotOurs(InstallChannel::Unknown);
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:192]: Tracked: KEN-831

Correct, and it is the population this issue is about. Every install predating the record classifies `NotOurs`, and the app writes a record only on the branch that already returned `Ours`, so an app-only user never bootstraps one.

Closing it means deciding when kendex may adopt a command it has no record for — the ownership question this PR settled deliberately after three rounds, and reopening it here is what the freeze on this PR exists to prevent. KEN-831 carries it, with the constraint that no rule may adopt a file on name and location alone, which is the P1 this PR closed.

Two one-command paths already bootstrap the record today: `kendex update` records the running binary up front, and re-running `install.sh` writes it. Neither is discoverable, which is the part KEN-831 has to fix rather than document.

File: crates/core/src/command_update.rs [hunk 14]
```diff
@@ -0,0 +1,378 @@
...
+pub fn bring_command_across(
+    env: &Env,
+    beside: &CommandBeside,
+    feed_url: &str,
+    release: &str,
+    target: &str,
+    public_key: &str,
+) -> Result<CommandHalf, String> {
+    let CommandBeside::Ours(path) = beside else {
+        return Ok(CommandHalf::Untouched);
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:227]: Fixed in af37782ce

Confirmed before changing anything: `move_the_command` calls a fresh discovery independent of what the card read, and `app_update_install` went on to download, install and restart on `Untouched` with nothing said anywhere.

`app_update_install` now takes what the card was showing and refuses the install when the fresh lookup disagrees — before the app half runs, so nothing is written, the app is untouched, and the card is still on screen to carry the refusal. No state crosses the restart because nothing restarts.

The comparison is on the notice rather than the classification: two `Ours` at different paths say the same nothing to a person, while `Ours` becoming `NotOurs` says something new, and the reverse would write over a file the card promised to leave alone. Both directions refuse. The store re-reads the command channel on failure, or the refusal's own sentence would be false and pressing Update now again would meet it forever.

No discovery rule, record semantics or ownership widening was touched — this PR is frozen on that mechanism.

One gap stated rather than hidden: a control that drops the `?` on `still_as_shown` inside `move_the_command` does not go red. That closure runs under `spawn_blocking` and needs a real updater, so it has no unit test here, as `app_update_install` itself has none. The wiring is one line adjacent to the tested function.

File: crates/core/src/command_update.rs [hunk 15]
```diff
@@ -0,0 +1,378 @@
...
+            "release {release} publishes no kendex command for {target}, so the command installed here would be left behind; nothing was updated"
+        ));
+    };
+    // Named as the command half. Read off a card that has just been
+    // pressed, a bare fetch or signature error says nothing about which of
+    // the two halves it came from.
+    let command = download(asset).map_err(command_half_failed)?;
+    command
+        .install_at(path, public_key)
+        .map_err(command_half_failed)?;
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:247]: Declined: this is the KEN-813 class, which was canceled on the same reasoning.

One correction to the premise first, because it changes the size of the window rather than its existence: the check is not the one from the card. `move_the_command` re-derives `command_beside_app` inside the `spawn_blocking` and runs `still_as_shown` against it before this function is entered, so the window is the three fetches below it, not the time a person left the card on screen.

What is left is a check and a write against a path with fetches between them. Re-reading the digest at the rename narrows that window; it cannot close it, because the writer who can win it is a writer who can already write that file as this user — and the same writer can replace the binary a second after the rename returns, with nothing left to check it. Buying a narrower race with a second read is the trade KEN-813 was canceled for.

Both reports are the same root cause on the same two lines, so this answers the copilot thread as well.

File: crates/core/src/command_update.rs [hunk 16]
```diff
@@ -0,0 +1,378 @@
...
+        return Err(format!(
+            "release {release} publishes no kendex command for {target}, so the command installed here would be left behind; nothing was updated"
+        ));
+    };
+    // Named as the command half. Read off a card that has just been
+    // pressed, a bare fetch or signature error says nothing about which of
+    // the two halves it came from.
+    let command = download(asset).map_err(command_half_failed)?;
+    command
+        .install_at(path, public_key)
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:246]: Declined: this is the KEN-813 class, which was canceled on the same reasoning.

One correction to the premise first, because it changes the size of the window rather than its existence: the check is not the one from the card. `move_the_command` re-derives `command_beside_app` inside the `spawn_blocking` and runs `still_as_shown` against it before this function is entered, so the window is the three fetches below it, not the time a person left the card on screen.

What is left is a check and a write against a path with fetches between them. Re-reading the digest at the rename narrows that window; it cannot close it, because the writer who can win it is a writer who can already write that file as this user — and the same writer can replace the binary a second after the rename returns, with nothing left to check it. Buying a narrower race with a second read is the trade KEN-813 was canceled for.

Both reports are the same root cause on the same two lines, so this answers the copilot thread as well.

File: crates/core/src/command_update/notice.rs [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    /// Nothing names an owner and no record proves the file is kendex's,
+    /// so there is no name to print and no command to offer.
+    Unknown,
+    /// Kendex's own command, where this app cannot write. `command` is
+    /// what carries it across with the privilege the app lacks.
+    NeedsPrivilege { path: String, command: String },
+}
+
+/// What `kendex update` is spelled as when it has to run as root.
+const ELEVATED_UPDATE: &str = "sudo kendex update";
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update/notice.rs:40]: Tracked: KEN-840.

Both halves are real, and neither is this PR's to take. `sudo` resolves a bare name against `secure_path` rather than this person's `PATH`, and `/usr/local/bin` — the case that reaches `NeedsPrivilege` at all, since it is the directory the app cannot write — is outside `secure_path` on some distributions and holds a different `kendex` on others. The second half is worse than the finding says: `sudo kendex update` runs as root, so `record_installed` writes root's record, the user's record stops matching, and the card drops to `Unknown`. The one command offered is a one-way door out of app-driven updates.

Not fixed here. This PR is under a scope freeze after three rounds reworking which command is kendex's to replace, and this is a fourth question about the same card. KEN-840 carries both halves, and names `names::quoted` as the spelling the path fix needs.

File: crates/core/src/command_update.rs [hunk 18]
```diff
@@ -0,0 +1,378 @@
...
+    probe: &dyn HostProbe,
+    candidates: &[PathBuf],
+    running: &[PathBuf],
+    installed: Option<&InstalledCommand>,
+) -> CommandBeside {
+    let running: Vec<PathBuf> = running.iter().map(|path| probe.resolve(path)).collect();
+    let record = installed.map(|record| (probe.resolve(&record.path), record.digest.as_str()));
+    // Last in the search, because everything ahead of it is what a shell
+    // resolves first, and that is the copy a person actually runs.
+    let searched = candidates.iter().chain(installed.map(|r| &r.path));
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/core/src/command_update.rs:171]: Declined: the order decides which command is refused, never which one is written.

`Ours` is returned only where `*path == resolved` and the digest of that same resolved path matches the record. There is no route through this chain by which a binary the record does not name becomes replaceable — the scenario ends at `NotOurs(InstallChannel::Unknown)`, and `a_record_of_one_command_does_not_vouch_for_another` is that state asserted: a first-found candidate the record does not name reads not-ours, whatever the digest of the bytes behind it.

The `~/.cargo/bin/kendex` case is already covered and passing. `a_recorded_command_no_other_route_reaches_is_still_found` drives exactly it — a record outside `PATH` and outside both installer directories, under the four system directories a Finder launch hands an app — and gets `Ours`. That test also pins its own premise: it asserts no other candidate names the file, because the record being searched is what makes it reachable at all.

What this finding adds to that fixture is a second, unrelated executable actually sitting at `~/.local/bin/kendex`. Stopping there is the deliberate half of the trade, and `command_beside_app`'s doc says why: the copy a person runs is the one their shell resolves first, and searching past it to find a replaceable one further down moves the half nobody runs and leaves the half they do. Putting the record ahead of the fallbacks buys the refusal you describe by paying for a wrong-target write in the case where the earlier `kendex` is the one on their shell's `PATH`.

It is also not silent. `CommandNotice::for_card` returns `Unknown` for that state and the card reads: "Update now updates the app only. Update the kendex command the way you installed it."

File: crates/cli/src/commands/update.rs [hunk 19]
```diff
@@ -1,10 +1,13 @@
 use std::path::{Path, PathBuf};
-use std::time::Duration;
 
+<<<<<<< HEAD
+use kendex_core::command_update::{fetch, record_command, replace_executable};
+=======
+use kendex_core::command_update::{download, record_command, record_installed};
+>>>>>>> c8d81d730 (fix(KEN-444): the record says which file, and where it can be written)
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/cli/src/commands/update.rs]: Declined: this describes 725607160, not the head. At e25475f5a there are no markers — a grep for conflict markers across `crates/`, `ui/` and `docs/` returns nothing, both crates build, and the macOS and Windows cargo lanes are green on this head.

Every resolution you name is the one in place. The app path keeps the download-only route so the release-digests check runs before `install_published`, and passes `shown` through to `move_the_command`. The CLI keeps `fetch`, `record_installed` and `replace_executable` from core, and line 170 records `binary`.

You caught a real thing: markers reached that intermediate commit through a `git add -A` during a rebase continue. They did not reach the branch head.

File: crates/app/src/app_update.rs [hunk 20]
```diff
@@ -223,19 +366,25 @@ pub async fn app_update_install(app: tauri::AppHandle) -> Result<(), String> {
         .await
         .map_err(|error| error.to_string())?
         .ok_or_else(|| "this build is already the latest release".to_owned())?;
+    let half = move_the_command(install, update.version.clone(), shown).await?;
```
[the remark this answers is not in the document]
Comment (bmethod) [reply crates/app/src/app_update.rs:369]: Declined: the PR is frozen to dispositions, and an endpoint test for `app_update_install` is not something this diff can add.

The gap is real and I have said so on this PR already. `app_update_install` is a `#[tauri::command]` that builds a real updater, reaches the network and restarts the app; there is no seam short of one that would have to be designed, and the same ceiling covers `install_published`, which KEN-820 shipped with the same limit. What is covered is each piece the call is made of: `bring_command_across`, `still_as_shown`, the notice comparison, and `app_half_failed`.

You are right that the ordering itself is what a test would pin, and right that moving the line would leave the suite green. That is the honest state of it, stated rather than papered over — the alternative on a frozen PR is a test built around a seam invented to be tested, which pins the seam and not the behaviour.
